test/reader/StorageReadable.t.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.7.6;
+pragma abicoder v2;
+
+import {Test, console} from "forge-std/Test.sol";
+import {StorageAccessibleWrapper} from "../../src/contracts/test/vendor/StorageAccessibleWrapper.sol";
+
+contract StorageReadableTest is Test {
+    StorageAccessibleWrapper instance;
+
+    function setUp() public {
+        instance = new StorageAccessibleWrapper();
+    }
+
+    function test_can_read_statically_sized_words() public {
+        instance.setFoo(42);
+        bytes memory actualBytes = instance.getStorageAt(instance.SLOT_FOO(), 1);
+        bytes memory expectedBytes = abi.encode(42);
+        assertEq(actualBytes, expectedBytes);
+    }
+
+    function test_can_read_fields_that_are_packed_into_single_storage_slot() public {
+        instance.setBar(7);
+        instance.setBam(13);
+        bytes memory actualBytes = instance.getStorageAt(instance.SLOT_BAR(), 1);
+        bytes memory expectedBytes = abi.encodePacked(new bytes(8), uint64(13), uint128(7));
+        assertEq(actualBytes, expectedBytes);
+    }
+
+    function test_can_read_arrays_in_one_go() public {
+        uint8 slot = instance.SLOT_BAZ();
+        uint256[] memory arr = new uint256[](2);
+        arr[0] = 42;
+        arr[1] = 1337;
+        instance.setBaz(arr);
+        bytes memory data = instance.getStorageAt(slot, 1);
+        uint256 length;
+        assembly {
+            length := mload(add(data, 0x20))
+        }
```

Comment:
The code is correct, but for simplicity it would be better if the reader didn't have to understand how byte vectors are encoded:

```suggestion
        uint256 length = abi.decode(data, (uint256));
```